src/client/Client.cc

```diff
@@ -11004,9 +11004,10 @@ void Client::_ll_get(Inode *in)
 
 int Client::_ll_put(Inode *in, uint64_t num)
 {
+  uint64_t old_ll_ref = in->ll_ref;
   in->ll_put(num);
   ldout(cct, 20) << __func__ << " " << in << " " << in->ino << " " << num << " -> " << in->ll_ref << dendl;
-  if (in->ll_ref == 0) {
+  if (old_ll_ref > 0 && in->ll_ref == 0) {
```

Comment:
This doesn't make much sense to me.

We just called ll_put on the inode, so if it's 0, then we _know_ that it was 1 before. ll_ref is protected by the Big Client Mutex, so it shouldn't be possible for another task to race in and change the ll_ref between the decrement and check.

That said, I did notice though is that in->ll_ref is unsigned and should probably be a signed value. Cleaning that up is a separate issue however.